### PR TITLE
Support http.server.request_count metric in otelhttp.

### PR DESCRIPTION
**Pull request by jeremyje**
Fixes https://github.com/open-telemetry/opentelemetry-go-contrib/issues/2481

### CODE REVIEW

**Review by hanyuancheung — APPROVED** [review]

File: instrumentation/net/http/otelhttp/handler.go [hunk 1]
```diff
@@ -202,6 +206,7 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	// Add metrics
 	attributes := append(labeler.Get(), semconv.HTTPServerMetricAttributesFromHTTPRequest(h.operation, r)...)
+	h.counters[RequestCount].Add(ctx, 1, attributes...)
```
Comment (Aneurysm9) [inline instrumentation/net/http/otelhttp/handler.go:209]: Is this necessary?  Can't the same information be extracted from the count of the server latency histogram?

**Review by Aneurysm9 — CHANGES_REQUESTED** [review]

File: instrumentation/net/http/otelhttp/handler.go (hunk 1 above)
Comment (RangelReale) [reply instrumentation/net/http/otelhttp/handler.go:209]: I'm using Dynatrace, at least in their UI I can't seem to find a way to get the count out of the `http.server.duration` metric, is it required to be available by opentelemetry?

**Comment by RangelReale** [comment]
Can this keep to be considered? I have no good way of getting the request count on my provider without this metric, which seems to be required by the spec.

**Comment by jeremyje** [comment]
I ended up deleting this PR unintentionally because I purged all my repositories. My list got way too long. 
If there's still interest in getting this merged I can recreate this PR but there was some pushback so it was never merged.

https://github.com/open-telemetry/opentelemetry-go-contrib/pull/2482/files still fortunately shows the diff.

cc: @Aneurysm9 @RangelReale @hanyuancheung

**Comment by dmathieu** [comment]
I'm not seeing this metric being required in the spec.
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md

**Comment by RangelReale** [comment]
> I'm not seeing this metric being required in the spec. https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md

Hmm you're right, I think I was confusing it with another thing. It isn't even mentioned there.

The only thing I'm afraid is adding it to my code, and in the future this being added to the library and it counting the request 2 times.

**Pull request closed without merging**